### PR TITLE
Add rainfall and urban growth examples

### PR DESCRIPTION
**Pull request by wang-boyu**

### CODE REVIEW

**Comment by rht** [comment]
How big are the .asc files? Is it customary to gzip them?

File: examples/rainfall/rainfall/model.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import uuid
+
+import numpy as np
+from mesa import Model
```
Comment (rht) [inline examples/rainfall/rainfall/model.py:4]: The minimum mesa version requirement is 1.0.0, and so you can do just `import mesa`. And do `mesa.Model`, etc, for the simplified namespace.

File: examples/rainfall/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+Rainfall Model
+==============
+
+An implementation of the [Rainfall Model](https://github.com/abmgis/abmgis/tree/master/Chapter06-IntegratingABMandGIS/Models/Rainfall) in Python, using [Mesa](https://github.com/projectmesa/mesa) and [mesa-geo](https://github.com/Corvince/mesa-geo).
```
Comment (rht) [inline examples/rainfall/README.md:4]: Should be projectmesa/mesa-geo.

File: examples/rainfall/rainfall/model.py (hunk 1 above)
Comment (wang-boyu) [reply examples/rainfall/rainfall/model.py]: Changed this as well.

**Comment by wang-boyu** [comment]
> How big are the .asc files? Is it customary to gzip them?

These files are less than or around 1 MB each. Not sure how it is done usually but most of the time I work with them directly.

**Comment by rht** [comment]
From looking at the .asc, it looks like it will be much smaller when compressed. I'd attempt to compress it.

**Comment by rht** [comment]
I know it won't affect `git clone`'s perf, because files are compressed in transit, but having a smaller repo size in people's laptop, for free (with gzip), would be nice.

**Comment by rht** [comment]
That's a significant size reduction from ~1 MB each.

**Comment by wang-boyu** [comment]
It's indeed much smaller. My only concern is that users won't be able to make a quick glance at those files without unzipping them first. But I've made the changes.

**Comment by rht** [comment]
You also removed the JPG result, which I personally prefer (so as not to bloat the repo), but @Corvince might prefer it included. But anyway, LGTM.

**Pull request merged**